claripy/ast/base.py

```diff
@@ -272,6 +272,11 @@ def __a_init__(self, op, args, variables=None, symbolic=None, length=None, simpl
                 tuple(a for a in self.annotations if not a.eliminatable and a.relocatable)
             ))).keys()
 
+            self.annotations = tuple(itertools.chain(
```

Comment:
I might be missing something, but doesn't the above already catch eliminatable=False, relocatable=True annotations?

Reply:
Yes that is correct, but they do not pass on the annotations from the `ast_args` to the resultant AST they only set the `self._uneliminatable` & `self._relocatable`. I am assuming that if `relocatable=True` and `eliminatable=False/True`, then the annotations from the arguments should be relocated to the annotations of the new AST(i.e. `self.annotations`).